config/v2_3_experimental/types/unit.go

```diff
@@ -120,12 +141,30 @@ func (d NetworkdDropin) Validate() report.Report {
 	return r
 }
 
-func validateUnitContent(content string) error {
+func validateUnitContent(content string) ([]*unit.UnitOption, error) {
 	c := strings.NewReader(content)
-	_, err := unit.Deserialize(c)
+	opts, err := unit.Deserialize(c)
 	if err != nil {
-		return fmt.Errorf("invalid unit content: %s", err)
+		return nil, fmt.Errorf("invalid unit content: %s", err)
+	}
+	return opts, nil
+}
+
+func (u Unit) validateInstallSection(contentSections []*unit.UnitOption) error {
+	isEnabled := u.Enable || (u.Enabled != nil && *u.Enabled)
+	if !isEnabled {
+		// install sections don't matter for not-enabled units
+		return nil
+	}
+	if len(contentSections) == 0 {
+		// install sections don't matter if it has no contents, e.g. it's being masked or just has dropins or such
```

Comment:
Empty if block? Based on the comment, did you intend for this to go in the for loop below, inside the `if section.Section == "Install"` check?

Reply:
whoops, meant for this to be an early return; fixed